api/src/service/domain/workflow/workflowitem_update.ts

```diff
@@ -61,25 +60,15 @@ export async function updateWorkflowitem(
     return new NotAuthorized(ctx, issuer.id, newEvent);
   }
 
-  try {
-    // Update a draft/copy of the workflowitem, leaving the original workflowitem
-    // unchanged for comparison:
-    workflowitem = withCopy(workflowitem, draft => {
-      // Check that the new event is indeed valid:
-      const result = WorkflowitemUpdated.apply(ctx, newEvent, draft);
-      if (Result.isErr(result)) {
-        throw new InvalidCommand(ctx, newEvent, [result]);
-      }
-
-      // Ignore the update if it doesn't change anything:
-      if (isEqual(workflowitem, result)) {
-        throw { newEvents: [], workflowitem };
-      }
+  // Check that the new event is indeed valid:
+  const result = withCopy(workflowitem, draft => WorkflowitemUpdated.apply(ctx, newEvent, draft));
```

Comment:
if we use the name `produce` everywhere else, we should probably change the remaining `withCopy`s to `produce` as well..

Reply:
Yes, we should discuss this. After reading the `immer` docs `produce` makes more sense for me + it matches the notation of the official doc.

Reply:
renamed it in 8a4175b638f8a7a460cb47f7c1226e729bbcf5d8